File: ios/Classes/FbAccessToken.m
#import "FbAccessToken.h"

@implementation FbAccessToken

+(NSDictionary*) parsedToken:(FBSDKAccessToken*)token {
  return token == nil ?
      @{}
    : @{ @"appId": [token appID],
         @"declinedPermissions": [[token declinedPermissions] allObjects],
         @"expirationTime": [NSNumber numberWithInteger:
                             [[token expirationDate] timeIntervalSince1970]],
         @"isExpired": [NSNumber numberWithBool:[token isExpired]],
         @"permissions": [[token permissions] allObjects],
         @"refreshTime": [NSNumber numberWithInteger:
                          [[token refreshDate] timeIntervalSince1970]],
         @"tokenString": [token tokenString],
         @"userId": [token userID],
         };
}

@end

File: android/src/main/java/com/sarpongkb/flutterfacebooksdk/FbAccessToken.java
package com.sarpongkb.flutterfacebooksdk;

import com.facebook.AccessToken;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class FbAccessToken {

  public static Map<String, Object> parsedToken(AccessToken token) {
    Map<String, Object> accessToken = new HashMap<>();

    if (token != null) {
      accessToken.put("appId", token.getApplicationId());
      accessToken.put("declinedPermissions", new ArrayList<>(token.getDeclinedPermissions()));
      accessToken.put("expirationTime", token.getExpires().getTime());
      accessToken.put("isExpired", token.isExpired());
      accessToken.put("permissions", new ArrayList<>(token.getPermissions()));
      accessToken.put("refreshTime", token.getLastRefresh().getTime());
      accessToken.put("tokenString", token.getToken());
      accessToken.put("userId", token.getUserId());
    }

    return accessToken;
  }
}
